### PR TITLE
Send new keep notifications a la BasicKeepEvent

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster

### CODE REVIEW

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessageThreadNotificationBuilder.scala [hunk 1]
```diff
@@ -281,11 +277,17 @@ class MessageThreadNotificationBuilderImpl @Inject() (
...
           thread = thread,
           threadStarter = basicUserById(thread.startedBy).externalId,
-          messageWithBasicUser = message,
-          unread = !message.user.right.toOption.exists(_.externalId == basicUserById(userId).externalId),
+          id = event.id,
+          eventTime = event.timestamp,
+          author = author,
+          text = DescriptionElements.formatPlain(event.header),
+          participants = participants.toSeq,
+          unread = basicUserById.get(userId).exists(_.externalId == author.externalId),
```
Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/MessageThreadNotificationBuilder.scala]: I think this needs a `!` in front

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala [hunk 2]
```diff
@@ -185,7 +190,7 @@ class KeepInternerImpl @Inject() (
       case (newKs, existingKs) => (newKs.map(_.keep), existingKs.map(_.keep))
     }
 
-    reportNewKeeps(newKeeps, libraryOpt.toSeq, context, notifyExternalSources = true)
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala:188]: @ryanpbrewster I'm pretty sure this boolean used to be relevant (would prevent us from spamming new keep notifs for imports), something changed and now it's always `true`, going to look into why that is.

**Comment by camhashemi** [comment]
@ryanpbrewster @leogrim updated. this PR deprecates system messages for old clients. we can do some work to keep them alive or after this we can kill `eliza.message.aux_data`.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala (hunk 2 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala:188]: I see that we killed the "don't spam on imports" logic a while back https://github.com/kifi/keepit/commit/ab8544948fcfbeaa446d5cfe1edfaf2c6f352381, I'm guessing that went hand-in-hand with when you guys implemented notification grouping.

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala:188]: So I think we good.

File: kifi-backend/modules/common/app/com/keepit/model/KeepRecipients.scala [hunk 3]
```diff
@@ -41,6 +41,8 @@ case class KeepRecipients(
     libraries = libraries ++ diff.libraries.added -- diff.libraries.removed
   )
 
+  def toDiff: KeepRecipientsDiff = KeepRecipientsDiff(DeltaSet.addOnly(users), DeltaSet.addOnly(libraries), DeltaSet.addOnly(emails))
```
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/model/KeepRecipients.scala:44]: simpler to write this as `this -- KeepRecipientsDiff.empty`

File: kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/internal/ElizaDiscussionController.scala [hunk 4]
```diff
@@ -118,11 +119,15 @@ class ElizaDiscussionController @Inject() (
...
-    import HandleKeepEvent._
-    implicit val context = heimdalContextBuilder().build
+  def modifyRecipientsAndSendEvent() = Action.async(parse.tolerantJson) { request =>
+    import ModifyRecipientsAndSendEvent._
     val input = request.body.as[Request]
-    discussionCommander.handleKeepEvent(input.keepId, input.commonEvent, input.basicEvent, input.source).map { _ => NoContent }
+    implicit val ctxt = HeimdalContext.empty
+    discussionCommander.modifyRecipientsForKeep(input.keepId, input.userAttribution, input.diff, input.source).andThen {
+      case Success((thread, diff)) => input.notifEvent.foreach { event =>
+        thread.allParticipantsExcept(input.userAttribution).foreach(userId => notifDeliveryCommander.notifyAddParticipants(userId, diff, thread, event))
```
Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/controllers/internal/ElizaDiscussionController.scala:128]: that's elegant, nice!

**Pull request merged**